lib/inputs/home_input.dart

```diff
@@ -0,0 +1,299 @@
+                          dropDownBilling = newValue;
+                        });
+                      },
+                      items: <String>[
+                        'Monthly',
+                        'Quarterly',
+                        'Half-Yearly',
+                        'Yearly'
+                      ].map<DropdownMenuItem<String>>((String value) {
+                        return DropdownMenuItem<String>(
+                          value: value,
+                          child: Text(value),
+                        );
+                      }).toList(),
+                      onSaved: (String selection) {
+                        newHomeInput.billingCycle = selection;
+                      },
+                    ),
+                  ),
+                ],
+              ),
+            ],
+          ),
+          // Consumption
+          Row(
+            children: <Widget>[
+              // Prompt
+              Column(
+                children: <Widget>[
+                  formTextBoxWidget(context, rowTitles[2]),
+                ],
+              ),
+              // Input
+              Column(
+                children: <Widget>[
+                  SizedBox(
+                    width: 120.0,
+                    height: 40.0,
+                    child: TextFormField(
+                      decoration: InputDecoration(
+                        labelText: 'kWh',
+                      ),
+                      validator: (value) {
+                        if (value.isEmpty) {
+                          return 'Please enter number';
+                        }
+                        return null;
+                      },
+                      keyboardType: TextInputType.number,
+                      onSaved: (String value) =>
+                          {newHomeInput.kwh = double.parse(value)},
+                    ),
+                  ),
+                ],
+              ),
+            ],
+          ),
+          // Submit Button
+          Row(
+            mainAxisAlignment: MainAxisAlignment.end,
+            children: <Widget>[
+              Column(
+                children: <Widget>[
+                  RaisedButton(
+                    onPressed: () {
+                      // Validate returns true if the form is valid, or false
+                      // otherwise.
+                      if (_formKey.currentState.validate()) {
+                        // If the form is valid, display a Snackbar.
+                        Scaffold.of(context).showSnackBar(
+                            SnackBar(content: Text('Processing Data')));
+                      }
+                      FormState form = _formKey.currentState;
+                      form.save();
+                      int calculatedEmission = calculateEmission(
+                          newHomeInput.billingCycle, newHomeInput.kwh);
+                      addEntry(
+                          calculatedEmission, DateTime.now(), 'Home Energy');
```

Comment:
Since 'Home Energy' is not registered as a source in the db, this won't work. But is ok for the moment.

Reply:
Yeah. Wasn't certain about that. I'll need to get familiar with DB implementation.